lib/secure_headers/railtie.rb

```diff
@@ -10,7 +10,7 @@ class Railtie < Rails::Railtie
                              'Public-Key-Pins', 'Public-Key-Pins-Report-Only']
 
       initializer "secure_headers.middleware" do
-        Rails.application.config.middleware.use SecureHeaders::Middleware
+        Rails.application.config.middleware.insert_before 0, SecureHeaders::Middleware
```

Comment:
For this feature to work correctly `SecureHeaders::Middleware` needs to come before the session middleware (or simply at the top of the rack stack).